In [2]:
2 + 3 + 9 # Sum

14

In [3]:
99 - 73 # Sub

26

In [4]:
23.54 * -1432 # Multipy

-33709.28

In [5]:
100 / 7 # Division

14.285714285714286

In [6]:
100 // 7 # Quotient

14

In [7]:
100 % 7 # Rest

2

In [8]:
5 ** 3 # Pow

125

In [9]:
((2 + 5) * (17 - 3)) / (4 ** 3) # Operation

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

You can use the empty cells below and add more cells if required.

## Solving multi-step problems using variables

01 - A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

In [23]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


02 - A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [59]:
costs_flying_airplane = 5000
ticket_per_person = 200
people_signed = 29

total_ticket_signed = ticket_per_person * people_signed
profit_percent = ((total_ticket_signed * 100) / costs_flying_airplane) - 100
total_profit = total_ticket_signed - costs_flying_airplane

print(f"Profit made made company in percent is {profit:.2f}%")
print(f"Profit made made company in value is $ {total_profit:.2f}")

Profit made made company in percent is 16.00%
Profit made made company in value is $ 800.00


## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [24]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [25]:
# Equality check - True
my_favorite_number == 1

True

In [30]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [28]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [31]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [32]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [33]:
# Less than check - True
my_favorite_number < 10

True

In [34]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [35]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [36]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [37]:
# Less than or equal check - True
3 + 6 <= 9

True

In [38]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

In [39]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >= 10

print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [40]:
my_favorite_number

1

In [41]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [42]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [43]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [44]:
True and False

False

In [45]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [47]:
a_neutral_number = 3

In [48]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [49]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [50]:
my_favorite_number < 0 or True

True

In [51]:
False or False

False

The not operator returns False if a condition is True and True if the condition is False.

In [52]:
not a_neutral_number == 3

False

In [53]:
not my_favorite_number < 0

True

In [54]:
not False

True

In [55]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses ( and ) to indicate the order in which logical operators should be applied.

In [56]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [57]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [58]:
not True and 0 < 1 or False and True

False

## Adding text styles using Markdown

Adding explanations using text cells (like this one) is a great way to make your notebook informative for other readers. It is also useful if you need to refer back to it in the future. You can double click on a text cell within Jupyter to edit it. In the edit mode, you'll notice that the text looks slightly different (for instance, the heading has a `##` prefix. This text is written using Markdown, a simple way to add styles to your text. Execute this cell to see the output without the special characters. You can switch back and forth between the source and the output to apply a specific style.

For instance, you can use one or more `#` characters at the start of a line to create headers of different sizes:

# Header 1

## Header 2

### Header 3

#### Header 4

To create a bulleted or numbered list, simply start a line with `*` or `1.`.

A bulleted list:

* Item 1
* Item 2
* Item 3

A numbered list:

1. Apple
2. Banana
3. Pineapple

You can make some text bold using `**`, e.g., **some bold text**, or make it italic using `*`, e.g., *some italic text.* You can also create links, e.g., [a link](https://jovian.ai). Images are easily embedded too:

![](https://i.imgur.com/3gjZMYK.png)

Another really nice feature of Markdown is the ability to include blocks of code. Note that code blocks inside Markdown cells cannot be executed.

```
# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

```

You can learn the full syntax of Markdown here: https://learnxinyminutes.com/docs/markdown/